scripts/globals/baitandswitch.lua

```diff
@@ -0,0 +1,203 @@
+-- Functions below used in quest: Bait and Switch
+-----------------------------------
+package.loaded["scripts/zones/Metalworks/TextIDs"] = nil;
+-----------------------------------
+require("scripts/zones/Metalworks/TextIDs");
+
+BnS_KILN = 1;
+BnS_REFINERY = 2;
+BnS_REPLICA = 3;
+
+function CheckBaitProgress(player)
+    local scope = 0;
+    local harp = 0;
+    local id = 0;
+    local snares = 0;
+    local charm = 0;
+    local watch = 0;
+    local costume = 0;
+
+    if (player:getMaskBit(player:getVar("Bait_and_Switch_Quest_Progress"),0) == false) then
+        scope = 1;
```

Comment:
don't need if block,
`scope = player:getMaskBit(player:getVar("Bait_and_Switch_Quest_Progress"),0) == false`

Reply:
same applies to the rest of this function

Reply:
not sure what you mean by this...

Reply:
he means you can just define scope as being the result of that check. Except he forgot to trim the =false off the end.
remove
```lua
    local scope = 0;
    if (player:getMaskBit(player:getVar("Bait_and_Switch_Quest_Progress"),0) == false) then
        scope = 1;
    end
```
replace with
```lua
    local scope =player:getMaskBit(player:getVar("Bait_and_Switch_Quest_Progress"),0);
```
coz getMaskBit returns true/false so your var will be one of those.

Reply:
ahhh... i get it... ok

Reply:
hmmm... looking at this i'm still needing to return scope as a number so it needs to be converted from boolean, (to be used as a param for cs) seems like whether i do it like i did or do it this way and convert it after... doesn't appear to be saving any steps... unless i'm looking at it wrong.

Reply:
@Wiggo32 is the cs params its being passed actually 11111 or did you want to set bits and then send the number those bits are into the cs param?

@takhlaq I don't remember what if anything lua has for turning bools to number, if I understood what you were telling him correct and he actually does need a number, I dunno how to help him there so *back to you, bob*.

Reply:
I need a number... just a 1 or a 0 based on whether the bit is true or false based on whether you've completed *that* aspect of the quest or not... there's nothing wrong with how it works already, it does exactly what it's supposed to do as is. I'm actually not really sure what we're trying to change about it. :man_shrugging:

Reply:
I shoulda just stayed outa it, thought i could help there. takhlaq will clarify when he gets back.

Reply:
no i didnt mistype my comment
something like:
```lua
function CheckBaitProgress(player)
    local ret = 
    { 
        player:getMaskBit(player:getVar("Bait_and_Switch_Quest_Progress"),0) == false, -- scope
        etc..
    }
    return ret
end
```
```lua
player:startEvent(0x038a, CheckBaitProgress(player)[item] and 1 or 0, 0, item, s_table.darha)
```
should work (needs testing)

Reply:
it already works... done testing. and i need every bit not just 0